Source/DafnyLanguageServer/Handlers/LanguageServerExtensions.cs

```diff
@@ -12,7 +13,11 @@ public static class LanguageServerExtensions {
     /// </summary>
     /// <param name="options">The language server where the handlers should be registered to.</param>
     /// <returns>The language server enriched with the dafny handlers.</returns>
-    public static LanguageServerOptions WithDafnyHandlers(this LanguageServerOptions options) {
+    public static LanguageServerOptions WithDafnyHandlers(this LanguageServerOptions options, DafnyOptions dafnyOptions) {
+      dafnyOptions.Plugins.ForEach(plugin =>
+        options = plugin is ConfiguredPlugin { Configuration: PluginConfiguration configuration } ?
+        configuration.WithPluginHandlers(options) : options
+      );
```

Comment:
```suggestion
      foreach(var plugin in dafnyOptions.Plugins) {
        options = plugin is ConfiguredPlugin { Configuration: PluginConfiguration configuration } ?
        configuration.WithPluginHandlers(options) : options
      );
```
For effectful loops, I prefer foreach statement rather than foreach expressions. Otherwise I had a hard time to figure out that you were augmenting options for the language server.

Reply:
Alright, I agree that this way might be easier to read. Changed this in https://github.com/dafny-lang/dafny/pull/5161/commits/5546b9cc595e43ca8666d7457900e044aee474c6.